# Taller Integrador — Módulo 1
## Máster en IA & Data Science | DevSeniorCode

**Nombre del estudiante:** Claudia Patricia Patiño Jaimes

**Fecha de entrega:** 26 de marzo de 2026  
**Repositorio GitHub:** https://github.com/pacatina/TallerFinalMod1MasterIA.git

---

> Este taller evalúa todo lo visto en el Módulo 1:  
> Python · Pandas · Git & GitHub · Documentación

### ¿Cómo entregar?
1. Completa todas las celdas marcadas con 
2. Asegúrate de que el notebook corra sin errores (Kernel → Restart & Run All)
3. Sube el notebook y el CSV a tu repositorio de GitHub
4. Incluye un README.md con tu nombre y una descripción breve del proyecto

---

## Contexto del Dataset

Trabajarás con el dataset **`salud_mundial.csv`**, que contiene indicadores de salud de **159 países**.

| Columna | Descripción |
|---|---|
| `pais` | Nombre del país |
| `region` | Continente o región geográfica |
| `nivel_ingresos` | Nivel económico: Low, Medium, High, Very High |
| `esperanza_vida` | Esperanza de vida al nacer (años) |
| `gasto_salud_usd` | Gasto per cápita en salud (USD/año) |
| `mortalidad_infantil` | Muertes por cada 1,000 nacidos vivos |
| `medicos_por_1000` | Médicos por cada 1,000 habitantes |
| `poblacion_urbana_pct` | % de población que vive en zonas urbanas |
| `obesidad_pct` | % de la población con obesidad |
| `diabetes_pct` | % de la población con diabetes |
| `tabaquismo_pct` | % de fumadores |
| `camas_hospital_por_1000` | Camas hospitalarias por 1,000 hab. |
| `vacunacion_pct` | % de cobertura de vacunación |

---

---
## Parte 1 — Python Fundamental
### (30 puntos)

Antes de usar Pandas, demostraremos que dominas las estructuras básicas de Python.

---

### 1.1 — Listas y operaciones básicas

Tienes la siguiente lista con la esperanza de vida de 10 países (en años).

In [7]:
# Lista dada — NO modificar
esperanzas_de_vida = [72.3, 81.5, 60.1, 78.9, 55.4, 83.2, 67.8, 74.0, 49.3, 79.6]

**Ejercicio 1.1** — Usando Python puro (sin Pandas ni funciones built-in como `sum()` o `max()`):
- Calcula el **promedio** de la lista usando un bucle `for`
- Encuentra el **valor máximo** usando un bucle `for`
- Encuentra el **valor mínimo** usando un bucle `for`

In [8]:
# Tu código aquí
promedio_vida = [72.3, 81.5, 60.1, 78.9, 55.4, 83.2, 67.8, 74.0, 49.3, 79.6]

#Variables 
sum = 0
max = promedio_vida [0]
min = promedio_vida [0]

#Recorre la lista
for cant in promedio_vida:
   sum += cant
   
   if cant > max:
    max = cant
    
   elif cant < min:
    min = cant

    
# Promedio
promedio = sum / len(promedio_vida)
print(f"{promedio:.2f}")

# Máximo
print(f"El promedio Maximo es : {max:.2f}")

# Mínimo
print(f"El promedio Maximo es : {min:.2f}")


70.21
El promedio Maximo es : 83.20
El promedio Maximo es : 49.30


### 1.2 — Diccionarios

Los diccionarios son fundamentales en Data Science para representar registros.

**Ejercicio 1.2** — Crea un diccionario llamado `pais_info` que represente a Colombia con los siguientes campos:
- `nombre`: "Colombia"
- `region`: "Americas"  
- `esperanza_vida`: 76.5
- `gasto_salud_usd`: 560.3
- `nivel_ingresos`: "High"

Luego:
1. Imprime solo las **claves** del diccionario
2. Imprime solo los **valores**
3. Agrega un nuevo campo: `vacunacion_pct` con valor 82.1
4. Imprime el diccionario completo

In [5]:
# Tu código aquí
#Diccionario
pais_info = {
    "nombre"            : "Colombia",
    "region"            : "Americas",
    "esperanza_vida"    : 76.5,
    "gasto_salud_usd"   : 560.3,
    "nivel_ingresos"    : "High"
}

#Imprimir solo las Claves
print("CLAVES:")
for claves in pais_info:
    print(claves)

#Imprimir solo los Valores
print("\nVALORES:")
print(f"{pais_info['nombre']}")
print(f"{pais_info['region']}")
print(f"{pais_info['esperanza_vida']}")
print(f"{pais_info['gasto_salud_usd']}")
print(f"{pais_info['nivel_ingresos']}")

#Agregar un nuevo campo

pais_info["vacunacion_pct"] = 82.1
print("\nNUEVO CAMPO:")
print(pais_info)

#Imprimir el diccionario completo
print("\nLISTA COMPLETA:")
print(f"País                :   {pais_info['nombre']}")
print(f"Región              :   {pais_info['region']}")
print(f"Esperanza de  Vida  :   {pais_info['esperanza_vida']}")
print(f"Gasto Salud         :   {pais_info['gasto_salud_usd']}")
print(f"Nivel ingresdo      :   {pais_info['nivel_ingresos']}")
print(f"Vacunacion PCT      :   {pais_info['vacunacion_pct']}")


CLAVES:
nombre
region
esperanza_vida
gasto_salud_usd
nivel_ingresos

VALORES:
Colombia
Americas
76.5
560.3
High

NUEVO CAMPO:
{'nombre': 'Colombia', 'region': 'Americas', 'esperanza_vida': 76.5, 'gasto_salud_usd': 560.3, 'nivel_ingresos': 'High', 'vacunacion_pct': 82.1}

LISTA COMPLETA:
País                :   Colombia
Región              :   Americas
Esperanza de  Vida  :   76.5
Gasto Salud         :   560.3
Nivel ingresdo      :   High
Vacunacion PCT      :   82.1


### 1.3 — Funciones

Las funciones nos permiten reutilizar lógica.

**Ejercicio 1.3** — Crea una función llamada `clasificar_esperanza_vida` que reciba un número (años) y retorne:
- `"Alta"` si es >= 75
- `"Media"` si es entre 60 y 74.9
- `"Baja"` si es < 60

Prueba la función con al menos 5 valores distintos e imprime los resultados.

In [9]:
# Tu código aquí

def clasificar_esperanza_vida():
    
    num = int(input("Escriba un numero: "))
    
    if num >= 75:
        return("Alta")
    elif num >=60 and num <= 74.9:
        return("Media") 
    else:
        return("Baja")
clasificar_esperanza_vida()

'Alta'

### 1.4 — Listas + Funciones + Bucles (integrador)

**Ejercicio 1.4** — Usa la función creada en 1.3 y la lista del ejercicio 1.1 para:
1. Crear una **nueva lista** llamada `clasificaciones` que contenga la clasificación de cada esperanza de vida
2. Imprimir cada par: `(valor, clasificación)` usando un bucle `for`

In [10]:
# Tu código aquí
#Lista punto 1.1
esperanzas_de_vida = [72.3, 81.5, 60.1, 78.9, 55.4, 83.2, 67.8, 74.0, 49.3, 79.6]

#Creacion Nueva Lista
clasificaciones = []

#Funcion creada en punto 1.3
def clasificar_esperanza_vida(num):
            
    if num >= 75:
        return("Alta")
    elif num >=60 and num <= 74.9:
        return("Media") 
    else:
        return("Baja")

#Creacion de lista de clasificaciones
for valor in esperanzas_de_vida:
    clasificacion = clasificar_esperanza_vida(valor)
    clasificaciones.append((valor, clasificacion))
#for para impresion de los pares
for valor, clasificacion in clasificaciones:
    print(valor, clasificacion)
    

72.3 Media
81.5 Alta
60.1 Media
78.9 Alta
55.4 Baja
83.2 Alta
67.8 Media
74.0 Media
49.3 Baja
79.6 Alta


---
## Parte 2 — Análisis con Pandas
### (50 puntos)

---

### 2.1 — Carga y exploración inicial

🔧 **Ejercicio 2.1** — Importa Pandas y carga el archivo `salud_mundial.csv`.  
Luego ejecuta los comandos necesarios para responder:
1. ¿Cuántas filas y columnas tiene el dataset?
2. ¿Qué tipos de datos tiene cada columna?
3. ¿Cuántos valores nulos hay por columna?
4. Muestra las primeras 8 filas

In [11]:
# Tu código aquí
import pandas as pd
import matplotlib.pyplot as plt

# Cargar el dataset
salud = pd.read_csv("salud_mundial.csv")
salud

# Dimensiones - Tiene 159 filas y 13 columnas
#salud.info()
print(salud.shape) 

# Tipos de datos dtypes: float64(10), str(3)
print(salud.dtypes)

# Valores nulos
salud.isnull().sum()

# Primeras 8 filas
salud.head(8)


(159, 13)
pais                           str
region                         str
nivel_ingresos                 str
esperanza_vida             float64
gasto_salud_usd            float64
mortalidad_infantil        float64
medicos_por_1000           float64
poblacion_urbana_pct       float64
obesidad_pct               float64
diabetes_pct               float64
tabaquismo_pct             float64
camas_hospital_por_1000    float64
vacunacion_pct             float64
dtype: object


,pais,region,nivel_ingresos,esperanza_vida,gasto_salud_usd,mortalidad_infantil,medicos_por_1000,poblacion_urbana_pct,obesidad_pct,diabetes_pct,tabaquismo_pct,camas_hospital_por_1000,vacunacion_pct
0,Afghanistan,Asia,Low,60.9,18.4,NaN,0.19,40.8,15.2,9.2,10.1,1.0,41.1
1,Albania,Europe,Medium,65.1,673.8,28.1,1.18,45.0,27.2,9.7,17.3,1.2,84.5
2,Algeria,Africa,Medium,74.5,525.3,28.2,0.59,47.3,15.2,7.6,20.8,1.3,71.4
3,Angola,Africa,Low,NaN,37.1,54.7,0.79,37.4,13.4,7.8,28.2,1.6,48.7
4,Argentina,Americas,High,79.4,1809.5,7.6,NaN,71.5,18.9,10.1,33.0,3.4,81.7
5,Armenia,Europe,Medium,69.2,797.5,24.2,2.44,67.5,10.2,12.8,32.0,2.3,71.1
6,Australia,Oceania,Very High,NaN,9964.1,NaN,4.42,87.2,20.6,10.3,19.7,7.3,93.8
7,Austria,Europe,Very High,83.7,3685.2,3.9,2.71,91.0,31.0,5.9,18.6,6.0,90.9


### 2.2 — Limpieza de datos

🔧 **Ejercicio 2.2** — El dataset tiene algunos valores nulos (¡datos reales siempre tienen este problema!).  
1. Rellena los valores nulos de columnas **numéricas** con la **media** de cada columna
2. Verifica que ya no queden nulos
3. Explica en una celda de texto: ¿Por qué usamos la media y no simplemente eliminamos esas filas?

In [12]:
# Tu código aquí

#Creamos un nuevo archivo para dejar el original como estaba inicialmente

salud_nuevo = salud.drop_duplicates()

#Creamos las medias de cada una de las columnas con nulos
media_esperanza_vida = salud_nuevo["esperanza_vida"].mean()
print(f"Media de esperanza de vida: {media_esperanza_vida:.2f}")
salud_nuevo["esperanza_vida"] = salud_nuevo["esperanza_vida"].fillna(media_esperanza_vida)

media_gasto_salud = salud_nuevo["gasto_salud_usd"].mean()
print(f"Media de gasto de salud USD: {media_gasto_salud:.2f}")
salud_nuevo["gasto_salud_usd"] = salud_nuevo["gasto_salud_usd"].fillna(media_gasto_salud)

media_mortalidad_infantil = salud_nuevo["mortalidad_infantil"].mean()
print(f"Media de mortalidad infantil: {media_mortalidad_infantil:.2f}")
salud_nuevo["mortalidad_infantil"] = salud_nuevo["mortalidad_infantil"].fillna(media_mortalidad_infantil)

media_medicos = salud_nuevo["medicos_por_1000"].mean()
print(f"Media de médicos por 1000: {media_medicos:.2f}")
salud_nuevo["medicos_por_1000"] = salud_nuevo["medicos_por_1000"].fillna(media_medicos)

media_obesidad = salud_nuevo["obesidad_pct"].mean()
print(f"Media de obesidad PCT: {media_obesidad:.2f}")
salud_nuevo["obesidad_pct"] = salud_nuevo["obesidad_pct"].fillna(media_obesidad)

media_diabetes = salud_nuevo["diabetes_pct"].mean()
print(f"Media de diabetes PCT: {media_obesidad:.2f}")
salud_nuevo["diabetes_pct"] = salud_nuevo["diabetes_pct"].fillna(media_diabetes)

media_tabaquismo = salud_nuevo["tabaquismo_pct"].mean()
print(f"Media de tabaquismo PCT: {media_tabaquismo:.2f}")
salud_nuevo["tabaquismo_pct"] = salud_nuevo["tabaquismo_pct"].fillna(media_tabaquismo)

media_camas_hospital = salud_nuevo["camas_hospital_por_1000"].mean()
print(f"Media de camas hospital por 1000: {media_camas_hospital:.2f}")
salud_nuevo["camas_hospital_por_1000"] = salud_nuevo["camas_hospital_por_1000"].fillna(media_camas_hospital)

media_vacunacion = salud_nuevo["vacunacion_pct"].mean()
print(f"Media de vacunación PCT: {media_vacunacion:.2f}")
salud_nuevo["vacunacion_pct"] = salud_nuevo["vacunacion_pct"].fillna(media_vacunacion)
print(f"Filas y Columnas despues de limpiar: {salud_nuevo.shape} ")
salud_nuevo.isnull().sum()


Media de esperanza de vida: 71.09
Media de gasto de salud USD: 2217.05
Media de mortalidad infantil: 28.92
Media de médicos por 1000: 1.98
Media de obesidad PCT: 19.22
Media de diabetes PCT: 19.22
Media de tabaquismo PCT: 22.92
Media de camas hospital por 1000: 3.05
Media de vacunación PCT: 77.62
Filas y Columnas despues de limpiar: (159, 13) 


pais                       0
region                     0
nivel_ingresos             0
esperanza_vida             0
gasto_salud_usd            0
mortalidad_infantil        0
medicos_por_1000           0
poblacion_urbana_pct       0
obesidad_pct               0
diabetes_pct               0
tabaquismo_pct             0
camas_hospital_por_1000    0
vacunacion_pct             0
dtype: int64

**Responde aquí** (doble clic para editar):

*¿Por qué usamos la media para rellenar nulos en vez de eliminar las filas?*

> Tu respuesta: Se rellena con la media por que independientemente del dato que tengamos el siempre va a buscar el punto medio del total de datos sin perder informacion importante que altere el resultado de la tabla

### 2.3 — Estadísticas descriptivas

🔧 **Ejercicio 2.3**
1. Muestra el resumen estadístico completo del dataset
2. ¿Cuál es la esperanza de vida **promedio** mundial?
3. ¿Cuál es el país con **mayor** gasto en salud? ¿Y el de **menor**?
4. ¿Cuál es la **mediana** de la mortalidad infantil?

In [13]:
# Tu código aquí

#1. Resumen estadistico completo

#salud_nuevo.info()

#2. Esperanza de vida promedio mundial
media_esperanza_vida = salud_nuevo["esperanza_vida"].mean()
print(f"El promedio de esperanza de vida mundial es: {media_esperanza_vida:.2f} %")

#3. Cual es el pais con mayor gasto en salud? y el menor?
#Pais con mayor gasto en salud

pais_max = salud_nuevo.loc[salud_nuevo["gasto_salud_usd"].idxmax()]
pais_min = salud_nuevo.loc[salud_nuevo["gasto_salud_usd"].idxmin()]

print(f"El pais como mayor gasto en salud es: ", pais_max["pais"], "-", pais_max["gasto_salud_usd"])
print(f"El pais como menor gasto en salud es: ", pais_min["pais"], "-", pais_min["gasto_salud_usd"])

#4. Cual es la mediana de la mortalidad infantil?

mediana_mortalidad_infantil = salud_nuevo["mortalidad_infantil"].median()
print(f"Mediana de mortalidad infantil: {mediana_mortalidad_infantil:.2f}")



El promedio de esperanza de vida mundial es: 71.09 %
El pais como mayor gasto en salud es:  Slovakia - 10311.8
El pais como menor gasto en salud es:  Afghanistan - 18.4
Mediana de mortalidad infantil: 23.70


### 2.4 — Filtrado y selección

🔧 **Ejercicio 2.4**
1. Filtra los países de **América** (`region == "Americas"`) y muestra solo las columnas: `pais`, `esperanza_vida`, `gasto_salud_usd`
2. Filtra los países con esperanza de vida **mayor a 80 años** — ¿cuántos son? ¿De qué regiones son?
3. Filtra los países con **nivel_ingresos == "Low"** y calcula su promedio de `mortalidad_infantil`
4. ¿Cuántos países tienen vacunación por encima del 90%?

In [14]:
# Tu código aquí

#1. Filtrar por paises de America mostrar columnas: pais, esperanza de vida y gastos salud
region_america_filtrado = salud_nuevo.loc[salud_nuevo["region"] == "Americas", ["pais", "esperanza_vida", "gasto_salud_usd"]]
print(region_america_filtrado)


                    pais  esperanza_vida  gasto_salud_usd
4              Argentina       79.400000          1809.50
11               Bolivia       75.800000           322.40
12                Brazil       74.100000          2913.70
16                Canada       78.900000          4019.60
17                 Chile       79.200000           854.50
19              Colombia       75.200000          2133.70
21            Costa Rica       76.300000          1772.80
23                  Cuba       78.900000          1804.20
26    Dominican Republic       71.085806           591.40
27               Ecuador       69.200000           638.80
29           El Salvador       71.900000           216.20
36             Guatemala       65.000000           442.80
37                 Haiti       51.200000            37.40
38              Honduras       74.200000           523.90
47               Jamaica       74.300000           279.50
59                Mexico       79.800000          2600.50
66            

In [15]:
#2. Filtra los países con esperanza de vida **mayor a 80 años** — ¿cuántos son? ¿De qué regiones son?

esperanza_vida_mayor_80 = salud_nuevo.loc[salud_nuevo["esperanza_vida"] >80]
print(esperanza_vida_mayor_80 ["region"].unique())


<StringArray>
['Europe', 'Asia', 'Americas']
Length: 3, dtype: str


In [16]:
#3. Filtra los países con **nivel_ingresos == "Low"** y calcula su promedio de `mortalidad_infantil`

ingresos_filtrado = salud_nuevo.loc[salud_nuevo["nivel_ingresos"] == "Low", ["nivel_ingresos"]]
prom_mortalidad = salud_nuevo["mortalidad_infantil"].mean()
print(ingresos_filtrado)
print(prom_mortalidad)


    nivel_ingresos
0              Low
3              Low
9              Low
14             Low
15             Low
20             Low
30             Low
34             Low
37             Low
51             Low
53             Low
56             Low
58             Low
61             Low
62             Low
63             Low
67             Low
68             Low
70             Low
79             Low
81             Low
83             Low
85             Low
90             Low
93             Low
94             Low
96             Low
99             Low
108            Low
109            Low
110            Low
115            Low
121            Low
123            Low
127            Low
130            Low
131            Low
134            Low
135            Low
138            Low
141            Low
143            Low
146            Low
149            Low
152            Low
157            Low
28.924832214765097


In [17]:
#4. ¿Cuántos países tienen vacunación por encima del 90%?

vacunacion_alta = salud_nuevo[salud_nuevo["vacunacion_pct"]>90]

print(len(vacunacion_alta))

39


### 2.5 — Agrupaciones

🔧 **Ejercicio 2.5** — Usa `groupby()` para:
1. Calcular la **esperanza de vida promedio** por `region`
2. Calcular el **gasto en salud promedio** por `nivel_ingresos`
3. Calcular el **número de países** por región
4. ¿Qué región tiene la mayor esperanza de vida promedio? ¿Y la menor?

In [18]:
# Tu código aquí

#1. Calcular la **esperanza de vida promedio** por `region`

salud_nuevo.groupby("region")["esperanza_vida"].mean()

region
Africa      63.513623
Americas    73.472531
Asia        70.704762
Europe      77.651950
Oceania     68.746452
Name: esperanza_vida, dtype: float64

In [19]:
#2. Calcular el **gasto en salud promedio** por `nivel_ingresos`

salud_nuevo.groupby("nivel_ingresos")["gasto_salud_usd"].mean()


nivel_ingresos
High         2003.016000
Low           126.881522
Medium        593.864894
Very High    6553.350000
Name: gasto_salud_usd, dtype: float64

In [20]:
#3. Calcular el **número de países** por región

salud_nuevo.groupby("region").size()
num_pais=salud_nuevo.groupby
#print(salud_nuevo[["region", "pais"]])
print(salud_nuevo.groupby("region")["pais"].apply(list))

region
Africa      [Algeria, Angola, Cameroon, Congo, Egypt, Ethi...
Americas    [Argentina, Bolivia, Brazil, Canada, Chile, Co...
Asia        [Afghanistan, Azerbaijan, Bangladesh, Cambodia...
Europe      [Albania, Armenia, Austria, Belgium, Bulgaria,...
Oceania      [Australia, New Zealand, Papua New Guinea, Fiji]
Name: pais, dtype: object


In [21]:
#4. ¿Qué región tiene la mayor esperanza de vida promedio? ¿Y la menor?

salud_nuevo.groupby("region")["esperanza_vida"].mean().sort_values(ascending=False)
salud_nuevo.head(1)

,pais,region,nivel_ingresos,esperanza_vida,gasto_salud_usd,mortalidad_infantil,medicos_por_1000,poblacion_urbana_pct,obesidad_pct,diabetes_pct,tabaquismo_pct,camas_hospital_por_1000,vacunacion_pct
0,Afghanistan,Asia,Low,60.9,18.4,28.924832,0.19,40.8,15.2,9.2,10.1,1.0,41.1


### 2.6 — Nueva columna calculada

🔧 **Ejercicio 2.6**
1. Crea una nueva columna llamada `categoria_vida` aplicando la función `clasificar_esperanza_vida` que creaste en la Parte 1 a la columna `esperanza_vida`
2. ¿Cuántos países caen en cada categoría? (usa `value_counts()`)
3. ¿Qué porcentaje del total representa cada categoría?

In [ ]:
# Tu código aquí

def clasificar_esperanza_vida(num):
    
    if num >= 75:
        return("Alta")
    elif num >=60 and num <= 74.9:
        return("Media") 
    else:
        return("Baja")


In [26]:
salud_nuevo["categoria_vida"] = salud_nuevo["esperanza_vida"].apply(clasificar_esperanza_vida)
nva_columna = salud_nuevo["categoria_vida"].value_counts().to_frame(name="Cantidad")
print(nva_columna)


                Cantidad
categoria_vida          
Media                 79
Alta                  63
Baja                  17


### 2.7 — Análisis libre (desafío)

**Ejercicio 2.7 (Opcional — puntos extra)** — Formula **una pregunta propia** sobre el dataset y respóndela con código.  

Ejemplos de preguntas que podrías hacer:
- ¿Existe relación entre el gasto en salud y la esperanza de vida?
- ¿Los países con más médicos tienen menor mortalidad infantil?
- ¿Qué región tiene mayor tasa de obesidad?

**Escribe tu pregunta:**

Para determinar en la poblacion urbana enfermedades que mas prevalecen analizar:

Cuales son los porcentajes de diabetes y tabaquismo en las regiones con mayor poblacion urbana?

In [29]:
# Tu código aquí

enfermedades = salud_nuevo.groupby("region").agg({"poblacion_urbana_pct":"mean","diabetes_pct":"mean", "tabaquismo_pct":"mean"}).reset_index()

enfermedades = enfermedades.sort_values(by="poblacion_urbana_pct", ascending=False)

print(enfermedades)


     region  poblacion_urbana_pct  diabetes_pct  tabaquismo_pct
3    Europe             76.779545      8.597428       21.220865
4   Oceania             66.325000      8.900000       24.850000
1  Americas             61.350000     10.261032       26.289156
2      Asia             56.990476      8.872462       23.779002
0    Africa             39.318605      8.204345       21.595769


---
## Parte 3 — Git & GitHub
### (20 puntos)

Esta parte se evalúa por el **estado real de tu repositorio** en GitHub.  
Documenta los comandos que usaste respondiendo las preguntas en las celdas de texto.

---

### 3.1 — Configuración del repositorio

**Responde** (doble clic para editar):

**¿Qué comandos usaste para crear e inicializar tu repositorio local?**
```bash
# Tu respuesta: 

#Utilice los siguientes comandos, git init, luego añadi los archivos con git add ., realice un git commit con el mensaje descriptivo de creacion y luego para finalizar un git push enviando todo al repositorio

```

**¿Cómo conectaste tu repositorio local con GitHub?**
```bash
# Tu respuesta:git remote add origin https://github.com/pacatina/TallerFinalMod1MasterIA.git

```

### 3.2 — Flujo de trabajo

**Responde:**

**¿Qué comando usaste para añadir los archivos al staging area?**
```bash
# Tu respuesta:

#El comando que use para añadir los archivos al staging area fue git add .

```

**¿Qué mensaje de commit escribiste? ¿Por qué elegiste ese mensaje?**
```bash
# Tu respuesta: 

#"Realizacion del punto 2.3 Estadisticas descriptivas" coloque este mensaje por que especifica que el cambio realizado fue que acababa de completar el punto 2.3 del trabajo, para facilitar el desarrollo del taller identificando que hice especificamente en este commit

```

**¿Cómo subiste los cambios a GitHub?**
```bash
# Tu respuesta: 

#Para subir los cambios de este paso hice: git add . para agregar todos los cambios, git commit -m "", especificando que se acababa de terminar la realizacion del punto 2.3  y por ultimo un git push que envia las actualizaciones al repositorio de GitHub 

```

### 3.3 — Buenas prácticas

**Responde:**

**¿Qué archivos debería incluir un `.gitignore` en un proyecto de Data Science? Menciona al menos 3 y explica por qué.**

Tu respuesta: 

Los archivos que deberia incluir un "gitignore" son: archivos innecesarios, que contengan información sensible, control de versiones, archivos demasiado pesados.  Se deben excluir por que pueden generar conflicto, se trata de mantener codigo limpio y ligero

**¿Cuál es la diferencia entre `git add .` y `git add nombre_archivo`?**

> Tu respuesta: 

git add . (actualiza todos los archivos que estan en la carpeta )
git add nombre_archivo (actualiza solo los ultimos avances del archivo mencionado )

---
## Parte 4 — Documentación (README)
### (Evaluada en GitHub)

Tu repositorio debe tener un archivo `README.md` que incluya:

- [ ] Título del proyecto
- [ ] Tu nombre
- [ ] Descripción del dataset (qué contiene, cuántos países, fuente)
- [ ] Cómo ejecutar el notebook (instrucciones paso a paso)
- [ ] Al menos 3 hallazgos o conclusiones del análisis
- [ ] Tecnologías usadas (Python, Pandas, Jupyter, Git)

---

### Checklist de entrega

Antes de hacer tu commit final, verifica:

- [ ] El notebook corre completo sin errores (Kernel → Restart & Run All)
- [ ] Todas las celdas están completadas
- [ ] Las celdas de texto tienen tus respuestas
- [ ] El archivo `salud_mundial.csv` está en el repositorio
- [ ] El `README.md` está completo
- [ ] El link al repositorio está en la celda de portada

---

## Criterios de evaluación

| Sección | Puntos |
|---|---|
| Parte 1 — Python | 30 |
| Parte 2 — Pandas | 50 |
| Parte 3 — Git | 20 |
| README en GitHub | 10 |
| Ejercicio libre 2.7 | +10 |
| **Total** | **110** |

